concent_api/core/subtask_helpers.py

```diff
@@ -11,133 +11,175 @@
-                response_type       = PendingResponse.ResponseType.SubtaskResultsSettled,
-                client_public_key   = subtask.requestor.public_key_bytes,
-                queue               = PendingResponse.Queue.ReceiveOutOfBand,
-                subtask             = subtask,
-            )
-        elif subtask.state == Subtask.SubtaskState.VERIFICATION_FILE_TRANSFER.name:  # pylint: disable=no-member
-
-            update_subtask_state(
-                subtask=subtask,
-                state=Subtask.SubtaskState.FAILED.name,  # pylint: disable=no-member
-            )
-            store_pending_message(
-                response_type=PendingResponse.ResponseType.SubtaskResultsRejected,
-                client_public_key=subtask.provider.public_key_bytes,
-                queue=PendingResponse.Queue.ReceiveOutOfBand,
-                subtask=subtask,
-            )
-            store_pending_message(
-                response_type=PendingResponse.ResponseType.SubtaskResultsRejected,
-                client_public_key=subtask.requestor.public_key_bytes,
-                queue=PendingResponse.Queue.ReceiveOutOfBand,
-                subtask=subtask,
-            )
-        elif subtask.state == Subtask.SubtaskState.ADDITIONAL_VERIFICATION.name:  # pylint: disable=no-member
-            task_to_compute = deserialize_message(subtask.task_to_compute.data.tobytes())
-
-            # Worker makes a payment from requestor's deposit just like in the forced acceptance use case.
-            payments_service.make_force_payment_to_provider(  # pylint: disable=no-value-for-parameter
-                requestor_eth_address=task_to_compute.requestor_ethereum_address,
-                provider_eth_address=task_to_compute.provider_ethereum_address,
-                value=task_to_compute.price,
-                payment_ts=get_current_utc_timestamp(),
-            )
-
-            update_subtask_state(
-                subtask=subtask,
-                state=Subtask.SubtaskState.ACCEPTED.name,  # pylint: disable=no-member
-            )
-            store_pending_message(
-                response_type=PendingResponse.ResponseType.SubtaskResultsSettled,
-                client_public_key=subtask.provider.public_key_bytes,
-                queue=PendingResponse.Queue.ReceiveOutOfBand,
-                subtask=subtask,
-            )
-            store_pending_message(
-                response_type       = PendingResponse.ResponseType.SubtaskResultsSettled,
-                client_public_key   = subtask.requestor.public_key_bytes,
-                queue               = PendingResponse.Queue.ReceiveOutOfBand,
-                subtask             = subtask,
-            )
-
-    logging.log_changes_in_subtask_states(
-        logger,
-        client_public_key,
-        clients_subtask_list.count(),
-    )
+        if subtask.state_enum == Subtask.SubtaskState.FORCING_RESULT_TRANSFER and subtask.requestor.public_key_bytes == client_public_key:
```

Comment:
this task might not be timed-out, but the function says "update **timed-out** subtasks"